libmozevent/mercurial.py

```diff
@@ -150,25 +150,26 @@ def apply_build(self, build):
         # Update the repo to base revision
         try:
             logger.info("Updating repo to revision {}".format(hg_base))
+            # See if the repo is clean
+            repo_status = self.repo.status(
+                modified=True, added=True, removed=True, deleted=True
+            )
+            if len(repo_status) != 0:
+                logger.warn(
+                    "Repo is dirty! Let's clean it first.",
+                    revision=hg_base,
+                    repo=self.name,
+                    repo_status=repo_status,
+                )
+                self.repo.update(rev="null", clean=True)
+
             self.repo.update(rev=hg_base, clean=True)
```

Comment:
We should first update, then check the status and, if the status is unclean, `hg up null` and then update again to the base revision

Reply:
I'm not sure, because the update it will fail since the repo is not clean.

Reply:
Isn't the bot currently failing at a later step than `update`?